library/src/main/java/com/parse/FacebookController.java

```diff
@@ -139,9 +139,9 @@ public void onError(FacebookException e) {
     return authData;
   }
 
-  public void setAuthData(Map<String, String> authData)
+  public void setAuthData(Map<String, String> authData, boolean isFacebookLoginScopeExternallyManaged)
       throws java.text.ParseException {
-    if (authData == null) {
+    if (authData == null && !isFacebookLoginScopeExternallyManaged) {
```

Comment:
But yes if we could just remove this logout all together, then I wouldn't need my flag at all.